dymos/trajectory/trajectory.py

```diff
@@ -417,129 +407,198 @@ def _configure_parameters(self):
+                control_name = vars[i][:-5] if classes[i] == 'polynomial_control_rate' else vars[i][:-6]
+                if units[i] is _unspecified:
+                    control_units = phases[i].polynomial_control_options[control_name]['units']
+                    time_units = phases[i].time_options['units']
+                    deriv = 1 if classes[i] == 'control_rate' else 2
+                    units[i] = get_rate_units(control_units, time_units, deriv=deriv)
+                shapes[i] = phases[i].polynomial_control_options[control_name]['shape']
+            elif classes[i] == 'parameter':
+                sources[i] = f'timeseries.parameters:{vars[i]}'
+                units[i] = phases[i].parameter_options[vars[i]]['units'] if units[i] is _unspecified else units[i]
+                shapes[i] = phases[i].parameter_options[vars[i]]['shape']
+            else:
+                rhs_source = phases[i].options['transcription']._rhs_source
+                sources[i] = f'{rhs_source}:{vars[i]}'
+                try:
+                    shapes[i], units[i] = get_source_metadata(phases[i]._get_subsystem(rhs_source),
+                                                              sources[i], user_units=units[i],
+                                                              user_shape=_unspecified)
+                except ValueError as e:
+                    raise ValueError(f'Invalid linkage:  Unable to find variable \'{vars[i]}\' in '
+                                     f'phase \'{phases[i].pathname}\' or its ODE.')
+
+        linkage_options['src_a'] = sources['a']
+        linkage_options['src_b'] = sources['b']
+        linkage_options['shape'] = shapes['b']
+        linkage_options['units'] = units['b']
+
+    def _expand_star_linkage_configure(self):
+        """
+        Finds the variable pair ('*', '*') and expands it out time time and all states if found.
+
+        Returns
+        -------
+        dict
+            The updated dictionary of linkages with '*' expanded to match time and all states at
+            a phase boundary.
+
+        """
+        for phase_pair, var_dict in self._linkages.items():
+            phase_name_a, phase_name_b = phase_pair
+
+            phase_b = self._get_subsystem(f'phases.{phase_name_b}')
+
+            for var_pair in var_dict.keys():
+                if tuple(var_pair) == ('*', '*'):
+                    options = var_dict[var_pair]
+                    self.add_linkage_constraint(phase_name_a, phase_name_b, var_a='time',
+                                                var_b='time', loc_a=options['loc_a'],
+                                                loc_b=options['loc_b'], sign_a=options['sign_a'],
+                                                sign_b=options['sign_b'])
+                    for state_name, state_options in phase_b.state_options.items():
+                        self.add_linkage_constraint(phase_name_a, phase_name_b, var_a=state_name,
```

Comment:
Feels kind of arbitrary for '*' to mean all the state vars on 'B'.  Seems like it should be all the state vars on 'A intersect B'.

Though, i don't know, maybe a warning if there are state vars on A not B and vice versa.

(This is assuming you are allowed to have different state vars on each phase, and I think i remember doing that in the battery problem.)

Reply:
So my reasoning is this:  If something is a state in phase b but was a control or output in phase a, then we should still want it to be continuous across that boundary.  For instance, phase A might pose the dynamics with "differential inclusions" where velocity is actually a control variable instead of a state.  If it is a state in phase-B, we'd want it to be continuous if possible.